MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/account/AccountRetrievalService.java

```diff
@@ -0,0 +1,163 @@
+package com.mapbox.mapboxandroiddemo.account;
+
+import android.app.IntentService;
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.support.annotation.Nullable;
+import android.util.Base64;
+import android.util.Log;
+
+import com.mapbox.mapboxandroiddemo.MainActivity;
+import com.mapbox.mapboxandroiddemo.model.usermodel.UserResponse;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+
+/**
+ * Background service which retrieves Mapbox Account information
+ */
+
+public class AccountRetrievalService extends IntentService {
+
+  private static final String BASE_URL = "https://api.mapbox.com/api/";
+  private static final String ACCESS_TOKEN_URL = "https://api.mapbox.com/oauth/access_token";
+
+  //  TODO: Fill in CLIENT_SECRET and before running app. Never commit changes to Github with CLIENT_SECRET filled in!
+  private static final String CLIENT_SECRET = "";
+
+  private String clientId;
+  private String redirectUri;
+  private String username;
+
+  public AccountRetrievalService() {
+    super("AccountRetrievalService");
+  }
+
+  @Override
+  public void onCreate() {
+    super.onCreate();
+  }
+
+  @Override
+  protected void onHandleIntent(@Nullable Intent intent) {
+    if (intent != null) {
+      String authCode = intent.getStringExtra("AUTHCODE");
+      clientId = intent.getStringExtra("clientId");
+      redirectUri = intent.getStringExtra("redirectUri");
+      getAccessToken(authCode);
+    } else {
+      Log.d("AccountRetrievalService", "onHandleIntent: intent == null");
+    }
+  }
+
+  private void getAccessToken(String code) {
+    OkHttpClient client = new OkHttpClient();
+
+    Request request = new Request.Builder()
+      .addHeader("User-Agent", "Android Dev Preview")
+      .addHeader("Content-Type", "application/x-www-form-urlencoded")
+      .url(ACCESS_TOKEN_URL)
+      .post(RequestBody.create(MediaType.parse("application/x-www-form-urlencoded"),
+        "grant_type=authorization_code&client_id=" + clientId + "&client_secret=" + CLIENT_SECRET
+          + "&redirect_uri=" + redirectUri + "&code=" + code))
+      .build();
+
+    client.newCall(request).enqueue(new okhttp3.Callback() {
+      @Override
+      public void onFailure(okhttp3.Call call, IOException exception) {
+
+      }
+
+      @Override
+      public void onResponse(okhttp3.Call call, okhttp3.Response response) throws IOException {
+
+        String json = response.body().string();
+
+        JSONObject data = null;
+        try {
+          data = new JSONObject(json);
```

Comment:
Referring to Retrofit's issue in branch `ls-switching-last-okhttp3-to-retrofit-for-login`.
As you can see in the logcat:

```
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug D/RetrievalService: onResponse: response received
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug D/RetrievalService: onResponse: json = Response{protocol=http/1.1, code=404, message=Not Found, url=https://api.mapbox.com/oauth/access_token/grant_type=authorization_code&client_id=%7BclientId%7D&client_secret=%7BclientSecret%7D&redirect_uri=%7BredirectUri%7D&code=%7Bcode%7D}
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug D/RetrievalService: onResponse: jsonString = retrofit2.OkHttpCall$NoContentResponseBody@79f9c5f
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err: org.json.JSONException: Value Response of type java.lang.String cannot be converted to JSONObject
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at org.json.JSON.typeMismatch(JSON.java:111)
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at org.json.JSONObject.<init>(JSONObject.java:160)
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at org.json.JSONObject.<init>(JSONObject.java:173)
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at com.mapbox.mapboxandroiddemo.account.AccountRetrievalService$1.onResponse(AccountRetrievalService.java:78)
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at retrofit2.ExecutorCallAdapterFactory$ExecutorCallbackCall$1$1.run(ExecutorCallAdapterFactory.java:68)
05-22 12:45:17.376 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at android.os.Handler.handleCallback(Handler.java:739)
05-22 12:45:17.377 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at android.os.Handler.dispatchMessage(Handler.java:95)
05-22 12:45:17.377 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at android.os.Looper.loop(Looper.java:148)
05-22 12:45:17.377 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at android.app.ActivityThread.main(ActivityThread.java:5417)
05-22 12:45:17.377 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
05-22 12:45:17.377 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
05-22 12:45:17.377 11938-11938/com.mapbox.mapboxandroiddemo.debug W/System.err:     at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

Retrofit was able to communicate with the given server, but the server could not find what was requested (`code=404, message=Not Found`) and `data` can't be created (see `org.json.JSONException: Value Response of type java.lang.String cannot be converted to JSONObject`).
So it seems that isn't a problem with Retrofit but with how the request is created/configured.